### PR TITLE
V13: Rework attempt pattern to use userkey

### PR DESCRIPTION
**Pull request by Zeegaan**
# Notes
- Rework all services to take a userKey instead of id
- Don't touch the obsolete ones to avoid breaking changes
- Notable changed IDataTypeContainerService.CreateAsync where now parent key does not have a default value, as this could trick people (it got me a few times) 🤣 
An argument could also be made to have them completely separate, so have a separate `CreateWithParentAsync`.

### CODE REVIEW

**Review by elit0451 — CHANGES_REQUESTED** [review]
Great work with all those changes 💪 🙌 

I've left a few comments and suggestions 😌 We also need to fix the pipelines before merging

File: src/Umbraco.Web.BackOffice/Controllers/TemplateController.cs [hunk 1]
```diff
@@ -26,20 +26,6 @@ public class TemplateController : BackOfficeNotificationsController
...
-        IShortStringHelper shortStringHelper,
-        IDefaultViewContentProvider defaultViewContentProvider)
-        : this(
-            StaticServiceProvider.Instance.GetRequiredService<ITemplateService>(),
-            umbracoMapper,
-            shortStringHelper,
-            defaultViewContentProvider,
-            fileService)
-    {
-    }
```
[the remark this answers is not in the document]
Comment (Zeegaan) [reply src/Umbraco.Web.BackOffice/Controllers/TemplateController.cs:41]: This consructor will be removed in v13 anyways, I just made the change to keep the template controller working for now 😁

File: src/Umbraco.Core/Services/DataTypeService.cs [hunk 2]
```diff
@@ -497,26 +507,29 @@ public void Save(IDataType dataType, int userId = Constants.Security.SuperUserId
                 throw new InvalidOperationException("Name cannot be more than 255 characters in length.");
             }
 
-            Attempt<IDataType, DataTypeOperationStatus> result = SaveAsync(
+            Guid currentUserKey = _userService.GetUserById(userId)?.Key ?? Constants.Security.SuperUserKey;
+            
+
```
Comment (elit0451) [inline src/Umbraco.Core/Services/DataTypeService.cs:512]:
```suggestion
            
```

**Review by elit0451 — APPROVED** [review]
Looks good 💪 Let's get this merged in!

**Pull request merged**